evennia/contrib/turnbattle/tb_range.py

```diff
@@ -0,0 +1,1356 @@
+    if mover.db.Combat_Range[target] > 2:
+        target.db.Combat_Range[mover] = 2
+        mover.db.Combat_Range[target] = 2
+                
+def approach(mover, target):
+    """
+    Manages a character's whole approach, including changes in ranges to other characters.
+    
+    Args:
+        mover (obj): The object moving
+        target (obj): The object to be moved toward
+        
+    Notes:
+        The mover will also automatically move toward any objects that are closer to the
+        target than the mover is. The mover will also move away from anything they started
+        out close to.
+    """
+    objects = mover.location.contents
+    
+    for thing in objects:
+        if thing != mover and thing != target:
+            # Move closer to each object closer to the target than you.
+            if mover.db.Combat_Range[thing] > target.db.Combat_Range[thing]:
+                distance_dec(mover, thing)
+            # Move further from each object that's further from you than from the target.
+            if mover.db.Combat_Range[thing] < target.db.Combat_Range[thing]:
+                distance_inc(mover, thing)
+    # Lastly, move closer to your target.
+    distance_dec(mover, target)
+    return 
+
+def withdraw(mover, target):
+    """
+    Manages a character's whole withdrawal, including changes in ranges to other characters.
+    
+    Args:
+        mover (obj): The object moving
+        target (obj): The object to be moved away from
+        
+    Notes:
+        The mover will also automatically move away from objects that are close to the target
+        of their withdrawl. The mover will never inadvertently move toward anything else while
+        withdrawing - they can be considered to be moving to open space.
+    """
+    objects = mover.location.contents
+    for thing in objects:
+        if thing != mover and thing != target:
+            # Move away from each object closer to the target than you, if it's also closer to you than you are to the target.
+            if mover.db.Combat_Range[thing] >= target.db.Combat_Range[thing] and mover.db.Combat_Range[thing] < mover.db.Combat_Range[thing]:
+                distance_inc(mover, thing)
+            # Move away from anything your target is engaged with
+            if target.db.Combat_Range[thing] == 0:
+                distance_inc(mover, thing)
+            # Move away from anything you're engaged with.
+            if mover.db.Combat_Range[thing] == 0:
+                distance_inc(mover, thing)
+    # Then, move away from your target.
+    distance_inc(mover, target)
+    return
+    
+def get_range(obj1, obj2):
+    """
+    Gets the combat range between two objects.
+    
+    Args:
+        obj1 (obj): First object
+        obj2 (obj): Second object
+        
+    Returns:
+        range (int or None): Distance between two objects or None if not applicable
+    """
+    # Return None if not applicable.
+    if not obj1.db.Combat_Range:
```

Comment:
Out of curiosity, there is no validation of `db.Combat_Range` in `distance_inc/dec` nor in `approach/withdraw`. If there is a risk of using objects without such an Attribute the check should probably either be extended to those too, or removed here for consistency (that is, if you use objects not valid for combat you should always expect a traceback).

Reply:
I do think this is a good function to have, though it was intuition more than anything rational that lead me to make it instead of referencing db.combat_range directly - I'll add it into those functions too.